### PR TITLE
(fix): Enhanced Tablet View: Service Queues Metrics Header Overhaul

### PR DESCRIPTION
**Pull request by Murithijoshua**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

This pull request introduces changes to the service-queues metrics header, specifically addressing the (add new service, add new queues service room & queues screen) buttons in tablet view. Given that a significant portion of our user base operates on tablets, the existing buttons often exceed the width of the screen. To mitigate this issue, I've incorporated ComboButton functionality, enabling users to access the buttons conveniently via a drop-down. See the attached video for the new change and screenshot image representing the actual issue. Thanks. 

## Screenshots

Screenshot_20240317_030300_Chrome

https://github.com/openmrs/openmrs-esm-patient-management/assets/37470288/13cb9b9e-c354-4464-a04d-19f8c53c42ed

### CODE REVIEW

**Review by chibongho — APPROVED** [review]
LGTM.

**Comment by ojwanganto** [comment]
cc @donaldkibet @makombe @denniskigen

File: packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx [hunk 1]
```diff
@@ -20,7 +23,45 @@ const MetricsHeader = () => {
   const navigateToQueueScreen = () => {
     navigate({ to: `${spaBasePath}/service-queues/screen` });
   };
+  if (isTablet || isPhone) {
+    return (
+      <div className={styles.metricsContainer}>
+        <span className={styles.metricsTitle}>{metricsTitle}</span>
+        <UserHasAccess privilege="Emr: View Legacy Interface">
```
Comment (denniskigen) [inline packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx:30]: Nit: Probably worth making this configurable. I don't know that this exact privilege works across different implementations. If it does, disregard my suggestion.

File: packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+import { navigate, UserHasAccess, useLayoutType } from '@openmrs/esm-framework';
 import { spaBasePath } from '../constants';
 import { SearchTypes } from '../types';
 import PatientSearch from '../patient-search/patient-search.component';
 import styles from './metrics-header.scss';
 
 const MetricsHeader = () => {
   const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const isPhone = useLayoutType() == 'phone';
```
Comment (denniskigen) [inline packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx:15]: Nit: I don't know that we cater to the phone viewport. Unless the associated breakpoints represent dimensions matching those of tablets in use in facilities. If not, maybe `tablet` alone is sufficient.

**Review by denniskigen — APPROVED** [review]
Thanks for the PR, @Murithijoshua. The UX looks much improved. I've left a couple of suggestions.

File: packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx (hunk 1 above)
Comment (mseaton) [reply packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx:30]: Agreed, but this isn't a change from the current behavior, so we should ticket separately I think

**Review by mseaton — APPROVED** [review]
Fine with me for the short term, though I generally think these buttons and such should come from extensions and not be wired in explicitly.

File: packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx (hunk 2 above)
Comment (Murithijoshua) [reply packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx:15]: from experience, we do have a few of data collection  tablets that are relatively smaller compared to standard Tablet.

Comment (ibacher) [reply packages/esm-service-queues-app/src/patient-queue-metrics/metrics-header.component.tsx:15]: @Murithijoshua Those would technically fall into the "unsupported" category.

**Pull request merged**